### PR TITLE
Safely update the contents of a file using rename in the pf::api

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

To avoid file corruption when updating configuration files through the api.
Use file rename instead of writing directly to the file.
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++

Bug Fixes
+++++++++
- Avoid the possibility of file corruption when updating configuration files through the api 
## Delete branch after merge

NO

### CODE REVIEW

File: lib/pf/api.pm [hunk 1]
```diff
@@ -522,13 +519,32 @@ sub distant_download_configfile : Public {
...
     return 1;
+}
+
+=head2 _safe_file_update
 
+This safely modifies the contents of a file using a rename
+
+=cut
+
+sub _safe_file_update {
```
Comment (julsemaan) [inline lib/pf/api.pm]: Belongs in pf::util much more than here

File: lib/pf/api.pm [hunk 2]
```diff
@@ -522,13 +519,32 @@ sub distant_download_configfile : Public {
...
+    syswrite $temp, $contents;
+    $temp->flush;
+    close $temp;
+    unless( rename ($temp->filename, $file) ) {
+        my $logger = pf::log::get_logger;
+        $logger->error("cannot save contents to $file '$!'");
+        die "cannot save contents to $file";
+    }
+    $temp->unlink_on_destroy(0);
+    `chown pf.pf $file`;
```
Comment (julsemaan) [inline lib/pf/api.pm]: That will not work as it also sync fingerbank files
You will need to use the reworked pfcmd fixpermissions

Comment (jrouzierinverse) [inline lib/pf/api.pm]: That feature depends on the pull request #879.

File: lib/pf/api.pm (hunk 1 above)
Comment (jrouzierinverse) [inline lib/pf/api.pm]: I will move into pf::util

File: lib/pf/api.pm (hunk 2 above)
Comment (julsemaan) [inline lib/pf/api.pm]: Yes, I would suggest to base off that PR and cite it as a dependency as both will be part of the same release

**Comment by jrouzierinverse** [comment]
Please review

**Comment by louismunro** [comment]
LGTM.

**Pull request merged**